includes/REST_API/Products_Controller.php

```diff
@@ -0,0 +1,489 @@
+            }
+          }
+        }
+      }
+    }
+  }
+}
+QUERY;
+
+		$response = wp_remote_post(
+			$url,
+			[
+				'headers' => [
+					'Content-Type'                      => 'application/graphql',
+					'X-Shopify-Storefront-Access-Token' => $access_token,
+				],
+				'body'    => $query,
+			]
+		);
+
+		if ( WP_Http::OK !== wp_remote_retrieve_response_code( $response ) ) {
+			return new \WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+		}
+
+		/**
+		 * Shopify GraphQL API response.
+		 *
+		 * @var ShopifyGraphQLResponse $result
+		 */
+		$result = json_decode( wp_remote_retrieve_body( $response ), true );
+
+		// TODO: Error handling.
+		if ( isset( $result['errors'] ) ) {
+			return new \WP_Error( 'rest_unknown', __( 'Error fetching products', 'web-stories' ), [ 'status' => 404 ] );
+		}
+
+		foreach ( $result['data']['products']['edges'] as $edge ) {
+			$product = $edge['node'];
+
+			$images = [];
+
+			foreach ( $product['images']['edges'] as $image_edge ) {
+				$image    = $image_edge['node'];
+				$images[] = [
+					'url' => $image['url'],
+					'alt' => $image['altText'],
+				];
+			}
+
+			$results[] = [
+				'productId'            => $product['id'],
+				'productTitle'         => $product['title'],
+				'productBrand'         => $product['vendor'],
+				// TODO: Maybe eventually provide full price range.
+				// See https://github.com/ampproject/amphtml/issues/37957.
+				'productPrice'         => $product['priceRange']['minVariantPrice']['amount'],
+				'productPriceCurrency' => $product['priceRange']['minVariantPrice']['currencyCode'],
+				'productImages'        => $images,
+				// TODO: Remove once optional, or find way to actually populate.
+				// See https://github.com/ampproject/amphtml/pull/37474.
+				'aggregateRating'      => [
+					'ratingValue' => 0,
+					'reviewCount' => 0,
+					'reviewUrl'   => $product['onlineStoreUrl'],
+				],
+				'productDetails'       => $product['description'],
+				'productUrl'           => $product['onlineStoreUrl'],
+			];
+		}
+
+		set_transient( $cache_key, wp_json_encode( $results ), $cache_ttl );
+
+		return rest_ensure_response( $results );
+	}
+
+
+	/**
+	 * Retrieves all WooCommerce products.
+	 *
+	 * @since 1.20.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	protected function get_items_woocommerce( $request ) {
+		$results = [];
+
+		/**
+		 * Products.
+		 *
+		 * @var array $products
+		 */
+		$products = wc_get_products(
+			[
+				'status'  => 'publish',
+				'limit'   => 100,
+				'orderby' => 'date',
+				'order'   => 'DESC',
+				's'       => $request['search'],
+			]
+		);
+
+		foreach ( $products as $product ) {
+			$images    = array_map(
+				static function( $image_id ) {
+					$url = wp_get_attachment_url( $image_id );
+					return [ 'url' => $url ];
```

Comment:
```suggestion
					$url = wp_get_attachment_url( $image_id );
					$alt = get_post_meta( $image_id, '_wp_attachment_image_alt', true );
					return [ 'alt' => $alt, 'url' => $url ];
```

Should we also add alt here as well?